app/views/partials/_review.html.erb

```diff
@@ -0,0 +1,16 @@
+<% if current_user.reviewed?(movie) %>
```

Comment:
@sdsunjay you could do something like this

in the Movie.rb file

``` rb
  def self.with_reviewed_by(user)
    Movie.left_joins(:reviews).where(Review.arel_table[:user_id].eq(user.id)).select([Arel.star, Review.arel_table[:id].count.as('reviewed')])
  end
```

then in the controller you could do `Movie.with_reviewed_by(current_user)`

and on each movie you'll have an integer value of how many reviews their are for the current user.

since its an integer, you'll have to check if `@movie.reviewed < 1` that way, this check will not do N+1 queries.

Reply:
Maybe @jprokay might have a better syntax for this, I'm not super great with arel these days and since rails 5 has been out, there are better helpers for it now.

Reply:
I was about to get this figured out with help from @excid3. Just load the user reviews and then search in those reviews. https://github.com/sdsunjay/movie-recommendations-with-ml/blob/ce0ff07c706299d69cd7476dad353b67519e26a0/app/views/partials/_review.html.erb#L4